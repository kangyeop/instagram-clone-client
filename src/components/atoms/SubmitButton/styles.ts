import styled from "styled-components";

interface IButton {
    canClick: boolean;
}

export const Button = styled.button<IButton>`
    cursor: pointer;

    outline: 0;

    border: 0;

    color: #0095f6;
    background-color: #ffffff;

    opacity: ${({ canClick }) => (canClick ? null : 0.3)};
`;
