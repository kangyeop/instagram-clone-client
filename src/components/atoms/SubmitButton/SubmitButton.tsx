import React from "react";
import { Button } from "./styles";

interface IProps {
    onClick: () => void;
    canClick: boolean;
    style?: React.CSSProperties;
}

const SubmitButton: React.FC<IProps> = ({ onClick, canClick, style, children }) => {
    return (
        <Button style={style} onClick={onClick} canClick={canClick} disabled={!canClick}>
            {children}
        </Button>
    );
};

export default SubmitButton;
