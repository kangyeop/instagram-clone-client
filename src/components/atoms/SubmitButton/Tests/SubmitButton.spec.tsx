import React from "react";
import { shallow } from "enzyme";
import SubmitButton from "../SubmitButton";

describe("Button component", () => {
    const clickFn = jest.fn();

    it("should be clickable", () => {
        const component = shallow(
            <SubmitButton onClick={clickFn} canClick>
                게시
            </SubmitButton>,
        );

        component.simulate("click");

        expect(clickFn).toHaveBeenCalled();
    });

    it("match snapshot", () => {
        const component = shallow(
            <SubmitButton onClick={clickFn} canClick>
                게시
            </SubmitButton>,
        );

        expect(component).toMatchSnapshot();
    });
});
