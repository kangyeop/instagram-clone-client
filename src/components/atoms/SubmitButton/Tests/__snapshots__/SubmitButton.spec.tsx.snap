// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component match snapshot 1`] = `
<styled.button
  canClick={true}
  disabled={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  게시
</styled.button>
`;
